### PR TITLE
Dockerpg

### PR DESCRIPTION
**Pull request by smnorris**

### CODE REVIEW

**Comment by smnorris** [comment]
I just updated the `overlay` command - it should be good but I haven't checked the outputs (still running). Also note that I haven't updated or checked the last step with Mapshaper, it probably needs some tweaks.

File: designatedlands.py [hunk 1]
```diff
@@ -0,0 +1,1283 @@
...
+        # set default n_processes to the number of cores available minus one
+        if self.config["n_processes"] == -1:
+            self.config["n_processes"] = multiprocessing.cpu_count() - 1
+
+        # don't try and use more cores than are available
+        elif self.config["n_processes"] > multiprocessing.cpu_count():
+            self.config["n_processes"] = multiprocessing.cpu_count()
+
+        self.db = pgdata.connect(self.config["db_url"])
+        self.db.ogr_string = f"PG:host={self.db.host} user={self.db.user} dbname={self.db.database} port={self.db.port}"
```
Comment (ateucher) [inline designatedlands.py:295]: I got an error with `gdal_rasterize` saying no password supplied. This fixed it for me:

```suggestion
        self.db.ogr_string = f"PG:host={self.db.host} user={self.db.user} dbname={self.db.database} password={self.db.password} port={self.db.port}"
```

File: designatedlands.py [hunk 2]
```diff
@@ -0,0 +1,1283 @@
...
+import pgdata
+
+
+LOG = logging.getLogger(__name__)
+
+
+# pick up db_url from $DATABASE_URL if available
+if "DATABASE_URL" in os.environ:
+    db_url = os.environ["DATABASE_URL"]
+else:
```
Comment (smnorris) [inline designatedlands.py:55]: Not sure why I left this here, the script picks up the db_url from the config file

File: designatedlands.py [hunk 3]
```diff
@@ -0,0 +1,1288 @@
...
+
+@cli.command()
+@click.argument("config_file", type=click.Path(exists=True), required=False)
+@verbose_opt
+@quiet_opt
+def dump(config_file, verbose, quiet):
+    """Dump output tables to file"""
+    set_log_level(verbose, quiet)
+    DL = DesignatedLands(config_file)
+    # overwrite output gpkg if it exists
```
Comment (ateucher) [inline designatedlands.py:1202]:
```suggestion
    # create output dir
    Path(DL.config["out_path"]).mkdir(parents=True, exist_ok=True)
    # overwrite output gpkg if it exists
```

**Pull request merged**